tests/ui_tools/test_buttons.py

```diff
@@ -314,8 +359,12 @@ def test_init(self, mocker):
         ],
     )
     def test_update_widget(
-        self, mocker, caption, expected_cursor_position, display_attr=None
-    ):
+        self,
+        mocker: MockerFixture,
+        caption: str,
+        expected_cursor_position: int,
+        display_attr: Optional[str] = None,
```

Comment:
If this is just `None` then the type is None, though it may warrant a comment to explain why we're only testing that value.

Reply:
This was typed the same way even in the method in the main code, wouldn't it be more consistent to type it the same way? This is something I've followed even in the annotations for `test_boxes.py` with the `state` parameter we pass into the autocomplete. I'm not sure what the best approach is here, but wouldn't it be easier to read as well if it's typed the same way in the tests, irrespective of whether only a subtype is being fulfilled by the cases?

Reply:
The type does give a suggestion that it should be expanded, so I can see this as an alternative :+1: